pkg/collector/remoteconfig/remote_config.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package remoteconfig schedule and unschedule integrations through remote-config
+// using the AGENT_INTEGRATIONS product
+package remoteconfig
+
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/secrets"
+	pkglog "github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Scheduler is the structure used to run checks with RC
```

Comment:
Nit: Might be worth making it more clear that we aren't actively running checks, we're just receiving configurations about checks that need to run _via remote-config_ and adding/removing those from the agent's scheduler as appropraite.